From 1e2264e3f67a26fe43039c1de61bf09a6cf8e41e Mon Sep 17 00:00:00 2001
From: Zhen Lei <thunder.leizhen@huawei.com>
Date: Fri, 8 Jan 2021 11:35:28 +0800
Subject: [PATCH 2674/4585] [Huawei] dt-bindings: arm: hisilicon: add binding
 for Hisilicon DJTAG

Offering: HULK
hulk inclusion
category: feature
bugzilla: 46869

-------------------------------------------------------------------------

Convert Hisilicon djtag bindings to json-schema. The original content in
txt is written by Tan Xiaojun <tanxiaojun@huawei.com>.

Signed-off-by: Zhen Lei <thunder.leizhen@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Patchwork Links: http://patchwork.huawei.com/project/hulk5.10/list/?series=13168
Signed-off-by: Zeng Heng <zengheng4@huawei.com>
---
 .../bindings/arm/hisilicon/djtag.yaml         | 78 +++++++++++++++++++
 1 file changed, 78 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/arm/hisilicon/djtag.yaml

diff --git a/Documentation/devicetree/bindings/arm/hisilicon/djtag.yaml b/Documentation/devicetree/bindings/arm/hisilicon/djtag.yaml
new file mode 100644
index 000000000000..c89b26f55309
--- /dev/null
+++ b/Documentation/devicetree/bindings/arm/hisilicon/djtag.yaml
@@ -0,0 +1,78 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/arm/hisilicon/djtag.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Hisilicon DJTAG r/w interfaces
+
+maintainers:
+  - Wei Xu <xuwei5@hisilicon.com>
+
+properties:
+  compatible:
+    enum:
+      - hisilicon,hip05-djtag
+      - hisilicon,hi1382-djtag
+      - hisilicon,sd6801-djtag
+      - hisilicon,djtag-v1
+      - hisilicon,djtag-v2
+      - hisilicon,djtag-v2-ex
+
+  syscon:
+    description: implement r/w interfaces via which sysctrl node.
+    $ref: /schemas/types.yaml#/definitions/phandle
+
+  djtag-mutex-access-protection:
+    description: |
+      Present if djtag need add mutex access protection to fix conflict with
+      firmware(like M3).
+    $ref: /schemas/types.yaml#/definitions/flag
+
+required:
+  - compatible
+  - syscon
+
+additionalProperties: false
+
+examples:
+  - |
+    /* If it have multi dies, set like below.
+     *
+     * We need sysctrl node for its register address and "syscon",
+     * but we don't implement sysctrl driver and don't care about
+     * what compatible it is.
+     */
+
+    /* for Hisilicon P660(hip05) or Hi1610 */
+    pa_sysctrl: system-controller@40010000 {
+        compatible = "hisilicon,peri-subctrl", "syscon";
+        reg = <0x40010000 0x10000>;
+    };
+
+    pc_sysctrl: system-controller@60010000 {
+        compatible = "hisilicon,peri-subctrl", "syscon";
+        reg = <0x60010000 0x10000>;
+    };
+
+    djtag0 {
+        compatible = "hisilicon,hip05-djtag";
+        syscon = <&pa_sysctrl>;
+    };
+
+    djtag1 {
+        compatible = "hisilicon,hip05-djtag";
+        syscon = <&pc_sysctrl>;
+    };
+
+    /* for Hisilicon Hi1382(hi1382) */
+    pc_sysctrl_f1010000: system-controller@f1010000 {
+        compatible = "hisilicon,peri-subctrl", "syscon";
+        reg = <0xf1010000 0x10000>;
+    };
+
+    djtag2 {
+        compatible = "hisilicon,hi1382-djtag";
+        syscon = <&pc_sysctrl_f1010000>;
+    };
+...
-- 
2.17.1

